mathics/algorithm/integrators.py

```diff
@@ -23,24 +26,24 @@
 def decompose_domain(interval, evaluation):
     if interval.has_form("System`Sequence", 1, None):
         intervals = []
-        for leaf in interval.leaves:
-            inner_interval = decompose_domain(leaf, evaluation)
+        for element in interval.elements:
+            inner_interval = decompose_domain(element, evaluation)
             if inner_interval:
                 intervals.append(inner_interval)
             else:
-                evaluation.message("ilim", leaf)
+                evaluation.message("ilim", element)
                 return None
         return intervals
 
     if interval.has_form("System`List", 3, None):
         intervals = []
-        intvar = interval.leaves[0]
+        intvar = interval.elements[0]
         if not isinstance(intvar, Symbol):
             evaluation.message("ilim", interval)
             return None
-        boundaries = [a for a in interval.leaves[1:]]
+        boundaries = [a for a in interval.elements[1:]]
         if any([b.get_head_name() == "System`Complex" for b in boundaries]):
```

Comment:
get_head() here?

Reply:
Actually it would be the equivalent of `Rest[]`. We don't have any special function for that, (see for example the eval method for Rest), although it would make sense if we did. And why this needs to be wrapped in a comprehension so I will try removing that.